browser/src/react-redux/App/Signup/SignupForm.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { Component } from "react";
+import axios from "axios";
+import Form from "./Form";
+import utility from "./signupUtility";
+
+class signupForm extends Component {
+  constructor() {
+    super();
+    this.state = {
```

Comment:
@Semiroundpizza8 @AuroraVaughn, what's your opinion on keeping state flat in React?

Reply:
I am assuming this is addressing React component state and not a Redux store.

 I try my best to keep things flat to reduce errors and increase readability.  React's set state is great but can get really convoluted when you have nested state. 

That said I have a soft spot in my heart for allowing address information to be nested in its own object. It could be the codebase I've worked in, but the address consists of several fields that almost always get passed and persisted as a whole.  I feel like having them in one ball of fields reduces the opportunity for other small mistakes. 

If your component has an address and just a couple of other values I keep it flat and use rest to gather address into an object in the render method. 

```
// formComplete in this example is the only other property on the state.

const {formComplete, ...address} = this.state
```

Reply:
Thanks Don and Aurora. I wasn't even aware nested objects were a bad idea in state. So, should a group of errors that are only delivered from the same server response be nested together, or is it best to be safe and just move them up a level?